pylibftdi/driver.py

```diff
@@ -72,36 +73,43 @@ class Driver(object):
     """
 
     # prefer libftdi1 if available. Windows uses 'lib' prefix.
-    _lib_search = {
-        'libftdi': ('ftdi1', 'libftdi1', 'ftdi', 'libftdi'),
-        'libusb': ('usb-1.0', 'libusb-1.0')
+    _lib_search: dict[str, list[str]] = {
+        'libftdi': ['ftdi1', 'libftdi1', 'ftdi', 'libftdi'],
+        'libusb': ['usb-1.0', 'libusb-1.0']
     }
 
-    def __init__(self, libftdi_search=None, **kwargs):
+    def __init__(self, libftdi_search: Optional[str]=None, **kwargs: dict[str, Any]) -> None:
         """
         :param libftdi_search: force a particular version of libftdi to be used
             can specify either library name(s) or path(s)
         :type libftdi_search: string or sequence of strings
         """
         if libftdi_search is not None:
-            self._lib_search['libftdi'] = libftdi_search
+            self._lib_search['libftdi'] = list(libftdi_search)
```

Comment:
this is broken:

```python
>>> libftdi_search = "path/to/libftdi"
>>> list(libftdi_search)
['p', 'a', 't', 'h', '/', 't', 'o', '/', 'l', 'i', 'b', 'f', 't', 'd', 'i']
```

should probably be (and yes, it would be nice if these were consistent):

```suggestion
            self._lib_search['libftdi'] = [libftdi_search]
```

but see comment above about handling both `str` and `List[str]` (as well as `None`)...

Reply:
Good catch, definitely not what I meant. I added a test case for this in `test_driver.py`. The initializer checks the type. If it's a string, create a list with a single element. If it's a list, replace the defaults with that list. If None, use the defaults. Raise a TypeError if it's none of those.